### DISCUSSION TITLE
"No module named run" when trying to start software to (later) debug

### DISCUSSION THREAD

**Discussion by dstromberg** [microsoft/debugpy#1029] [Q&A]
Hi folks.

I'm working on a (belated) Python 2 -> Python 3 port of some internal-only software.

We're starting things up using docker-compose.

With Python 2, this works, with and without debugging:
/usr/bin/python2 -m debugpy --listen 0.0.0.0:5678 /app/app/run.py runserver -h 0.0.0.0 -p 5000

With python 3, this runs the software, but does not allow the kind of debugging most of us use:
/usr/local/bin/tact-python /app/app/run.py runserver -h 0.0.0.0 -p 5000

With python 3, this fails to run the sofware, with or without debugging:
/usr/local/bin/tact-python -m debugpy --listen 0.0.0.0:5678 /app/app/run.py runserver -h 0.0.0.0 -p 5000
...giving the error message:
No module named run

/usr/local/bin/tact-python is a symlink to a local build of CPython 3.10.

We are using debugpy 1.4.3 with CPython 2, and are attempting to use debugpy 1.6.3 with CPython 3.  debugpy 1.4.3 gives that same error message with CPython 3.10.

I probably should be a little more clear. My question is: How can we start our application under debugpy for both use with a debugger, and for use without a debugger, for both CPython 2 and CPython 3?

Thanks :)

**Comment by fabioz** [comment]
Related to the `No module named run` message, I can't really reproduce it here (it seems to be in your app).

If you add `--wait-for-client` and attach prior to the run, can you get further?

If you're still blocked with that, is it possible you you to provide the logs for the run? See: https://github.com/microsoft/debugpy/wiki/Enable-debugger-logs for details on how to do that.
